### PR TITLE
[Docs] Client count doc updates for ACME update in 1.14

### PR DESCRIPTION
**Pull request by schavis**
Changes:
- Rewrote "Client count" overview to focus on a discussion of how entity assignment works and added ACME-related text.
- Re-titled "Client count" page to "Clients and entities".
- Moved content that focuses on the calculations to a new document called "Client count calculation" and rewrote the text to reference language and concepts in the client count overview.
- Moved the best practices discussion for CI/CD pipelines to a new best practices  document (with minimal edits) called "Best practices for CI/CD pipelines" under /secrets/pki/best-practices/cicd-pipelines"
- Updated TOC

### CODE REVIEW

**Review by cipherboy — APPROVED** [review]
Thank you very much! The ACME client count updates look correct to me :-)

File: website/content/docs/concepts/client-count/counting.mdx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+For example:
+
+- Two non-entity tokens under the same namespace, with the same alias name and
+  policy assignment receive the same entity assignment and are only counted
+  **once**.
+- Two authentication requests from a single ACME client for the same certificate
+  identifiers from different mounts receive the same entity assignments and
+  are counted **once**.
+- An application authenticating with AppRol receive the same entity assignment
```
Comment (brianshumate) [inline website/content/docs/concepts/client-count/counting.mdx:57]:
```suggestion
- An application authenticating with AppRole receive the same entity assignment
```

File: website/content/docs/concepts/client-count/index.mdx [hunk 2]
```diff
@@ -1,323 +1,185 @@
...
-|           20               |         15            |      100%    |
-|           20               |         100           |      100%    |
-|           20               |         1000          |      100%    |
-|           20               |         10000         |      70%     |
-|           200              |         10000         |      94%     |
-|           2000             |         10000         |      98%     |
+Client identity type       | AuthN method
+-------------------------- | ---------------------------------
+Externally managed or SSO  | Active Directory, LDAP, OIDC, JWT, GitHub, username+password
+Platform- or server-based  | Kubernetes, AWS, GCP, Azure, PKI, Cloud Foundry
```
Comment (cipherboy) [inline website/content/docs/concepts/client-count/index.mdx:35]:
```suggestion
Platform- or server-based  | Kubernetes, AWS, GCP, Azure, Cert, Cloud Foundry
```

PKI is a secrets engine, not an auth engine; Cert is the auth engine.

File: website/content/docs/concepts/client-count/index.mdx [hunk 3]
```diff
@@ -1,323 +1,185 @@
...
+Self                       | AppRole, tokens with no associated authN path or role
+ACME                       | Certificate
```
Comment (cipherboy) [inline website/content/docs/concepts/client-count/index.mdx:37]:
```suggestion
```

File: website/content/docs/concepts/client-count/index.mdx [hunk 4]
```diff
@@ -1,323 +1,185 @@
...
+1. **Servers and platforms** like VMs, Docker containers, or Kubernetes pods.
+1. **Orchestrators** like Nomad, Terraform, Ansible, ACME, and other continuous
+   integration / continuous delivery (CI/CD) pipelines.
+1. **Vault agents and proxies** that act on behalf of an application or
+   microservice.
 
-![Vault Client Types](https://www.datocms-assets.com/2885/1617325030-vault-clients.png)
+## Identity and entity assignment
+
+When clients authenticate, Vault assigns them a **client identity type**:
```
Comment (cipherboy) [inline website/content/docs/concepts/client-count/index.mdx:30]: Mohan: This comment from the original made sense here about why:

> There can be many different types of clients that authenticate and communicate with Vault using one of the above identities, including:

**Review by brianshumate — CHANGES_REQUESTED** [review]
Looks great! I found just one typo and good to go from me.

File: website/content/docs/concepts/client-count/index.mdx [hunk 5]
```diff
@@ -1,323 +1,185 @@
...
-
-| Current Month Clients      | Total Months' Clients | Accuracy     |
-| :---                       |    :----:             |        ---:  |
-|           20               |         15            |      100%    |
-|           20               |         100           |      100%    |
-|           20               |         1000          |      100%    |
-|           20               |         10000         |      70%     |
-|           200              |         10000         |      94%     |
-|           2000             |         10000         |      98%     |
+Client identity type       | AuthN method
```
Comment (cipherboy) [inline website/content/docs/concepts/client-count/index.mdx:32]:
```suggestion
Authorization Source       | AuthN method
```

**Review by brianshumate — APPROVED** [review]

**Pull request merged**